policy/modules/services/container.te

```diff
@@ -0,0 +1,880 @@
+policy_module(container, 1.0.0)
+
+## <desc>
+##	<p>
+##	Allow containers to manage cgroups.
+##	This is required for systemd to run inside
+##	containers.
+##	</p>
+## </desc>
+gen_tunable(container_manage_cgroup, false)
+
+## <desc>
+##	<p>
+##	Allow container engines to mount on all non-security files.
+##	</p>
+## </desc>
+gen_tunable(container_mounton_non_security, false)
+
+## <desc>
+##	<p>
+##	Allow containers to use NFS filesystems.
+##	</p>
+## </desc>
+gen_tunable(container_use_nfs, false)
+
+## <desc>
+##	<p>
+##	Allow containers to use CIFS filesystems.
+##	</p>
+## </desc>
+gen_tunable(container_use_samba, false)
+
+########################################
+#
+# Declarations
+#
+
+attribute container_domain;
+attribute container_net_domain;
+attribute container_user_domain;
```

Comment:
I don't understand what these represent.

Reply:
I'm unsure in general about "user" in this policy. Is it a user container vs system container?

Reply:
`container_net_domain` can be used to create container domains with/without network access. A container domain that has `container_net_domain` will be permitted to access the network. I want to keep this separation because it means we can allow policy writers to customize their container types in their local policy, i.e. with [Udica](https://github.com/containers/udica).

>Is it a user container vs system container?

Yes. All references to "user" in the policy are for user containers while "system" are for system containers. I agree I don't think it is very clear, so I can add some additional comments in the policy to make that distinction unless you think it should be changed.

Reply:
Perhaps it would be clearer if "system" and "user" were in the type names.

Reply:
At the moment `container_t` is a common type for both system and user containers. Do you think this should be split to `container_system_t` and `container_user_t`?